documentation/adr/0004-trip-modifications-mqtt-topic-structure.md

```diff
@@ -0,0 +1,134 @@
+# 4. Trip Modifications MQTT Topic Structure
+
+Date: 2024-07-04
+
+## Status
+
+Draft
+
+## Context
+
+[With the creation of #ADR0003](./0003-use-mqtt-to-publish-trip-modifications.md),
+we are trying to decide on the structure for _how_ we send and persist these 
+[Trip Modifications](https://gtfs.org/realtime/reference/#message-tripmodifications) between Skate and Transit Data.
+
+We know that we'll need these messages to be available if the Transit Data
+service is restarted.
+
+## Decision
+
+We'll use a
+[topic hierarchy](https://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices/)
+which allows us to manage individual GTFS-RT messages and update them, and does
+not require that the Transit Data service has some way to persist state between
+restarts.
+
+### Topic Hierarchy
+We'll nest all of our Trip Modifications related data under the
+[MQTT "topic level"](https://www.hivemq.com/blog/mqtt-essentials-part-5-mqtt-topics-best-practices/)
+`trip_modifications`, which will be under our "topic prefix"
+configured in the DevOps repo for each environment.
+
+For Example: (spaces used for alignment, and are not part of the topic)
+```
+skate/prod     /trip_modifications/
+skate/dev      /trip_modifications/
+skate/dev-blue /trip_modifications/
+skate/dev-green/trip_modifications/
+```
+
+
+For each new and unique trip modification, we'll generate a unique ID
+(likely `UUID` using `Ecto.UUID.generate()`)
```

Comment:
```suggestion
```

Nit: I don't think we need to (or want to) specify how we'll generate the ID's. So long as they're unique, I think we're good.